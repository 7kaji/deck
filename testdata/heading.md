# Normal H1 only slide

This is a slide with only H1 headings.

# Another H1 slide

Content for the second H1 slide.

---

# H1 and H2 slide

This is an H1 title.

## H2 subtitle

Content under H2.

## Another H2 subtitle

More content under H2.

# Next H1 slide

Another main slide.

---

## H2 only slide

This is a slide with only H2 headings.

## Another H2 slide

Content for the second H2 slide.

## Third H2 slide

More H2 content.

---

## H2 and H3 slide

This is an H2 title.

### H3 subtitle

Content under H3.

### Another H3 subtitle

More content under H3.

## Next H2 slide

Another H2 slide.

### H3 under next H2

Content under this H3.

---

## H2, H3 and H4 slide

This is an H2 title.

### H3 subtitle

Content under H3.

#### H4 content

This is H4 level content.

#### Another H4 content

More H4 level content.

### Another H3 subtitle

More content under H3.

#### H4 under this H3

H4 content here.

## Next H2 slide

Another H2 slide with mixed levels.

---

### H3 and H4 slide

This is an H3 title.

#### H4 subtitle

Content under H4.

#### Another H4 subtitle

More content under H4.

### Next H3 slide

Another H3 slide.

#### H4 under next H3

Content under this H4.

---

## H2 and H4 slide (skipping H3)

This is an H2 title.

#### H4 content directly under H2

This H4 is directly under H2, skipping H3 level.

#### Another H4 content

More H4 content without H3.

## Next H2 slide

Another H2 slide.

#### Direct H4 again

H4 content directly under H2.
