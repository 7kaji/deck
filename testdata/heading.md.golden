[
  {
    "layout": "",
    "titles": [
      "Normal H1 only slide",
      "Another H1 slide"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is a slide with only H1 headings."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content for the second H1 slide."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 1
  },
  {
    "layout": "",
    "titles": [
      "H1 and H2 slide",
      "Next H1 slide"
    ],
    "subtitles": [
      "H2 subtitle",
      "Another H2 subtitle"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is an H1 title."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content under H2."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "More content under H2."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Another main slide."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 1
  },
  {
    "layout": "",
    "titles": [
      "H2 only slide",
      "Another H2 slide",
      "Third H2 slide"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is a slide with only H2 headings."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content for the second H2 slide."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "More H2 content."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 2
  },
  {
    "layout": "",
    "titles": [
      "H2 and H3 slide",
      "Next H2 slide"
    ],
    "subtitles": [
      "H3 subtitle",
      "Another H3 subtitle",
      "H3 under next H2"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is an H2 title."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content under H3."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "More content under H3."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Another H2 slide."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content under this H3."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 2
  },
  {
    "layout": "",
    "titles": [
      "H2, H3 and H4 slide",
      "Next H2 slide"
    ],
    "subtitles": [
      "H3 subtitle",
      "Another H3 subtitle"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is an H2 title."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content under H3."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "H4 content",
                "bold": true
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "This is H4 level content."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Another H4 content",
                "bold": true
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "More H4 level content."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "More content under H3."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "H4 under this H3",
                "bold": true
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "H4 content here."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Another H2 slide with mixed levels."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 2
  },
  {
    "layout": "",
    "titles": [
      "H3 and H4 slide",
      "Next H3 slide"
    ],
    "subtitles": [
      "H4 subtitle",
      "Another H4 subtitle",
      "H4 under next H3"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is an H3 title."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content under H4."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "More content under H4."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Another H3 slide."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Content under this H4."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 3
  },
  {
    "layout": "",
    "titles": [
      "H2 and H4 slide (skipping H3)",
      "Next H2 slide"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is an H2 title."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "H4 content directly under H2",
                "bold": true
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "This H4 is directly under H2, skipping H3 level."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Another H4 content",
                "bold": true
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "More H4 content without H3."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Another H2 slide."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Direct H4 again",
                "bold": true
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "H4 content directly under H2."
              }
            ]
          }
        ]
      }
    ],
    "title_heading_level": 2
  }
]